juju/controller.py

```diff
@@ -39,9 +40,11 @@ def __init__(self, loop=None,
         """Connect to the current Juju controller.
 
         """
-        self.connection = (
-            await connection.Connection.connect_current_controller(
-                max_frame_size=self.max_frame_size))
+        jujudata = connection.JujuData()
```

Comment:
What's the reasoning behind this change? Generally I think we have tried to read juju data from disk as little as possible, and only when there was no other option, because relying on it basically means that you need the cli installed in order to use the python lib.

Reply:
The juju data was already being read in `Connection.connect_current_controller` but then the controller name wasn't being saved on the `Controller` instance.  This was actually from an earlier approach to fixing the test issue I'll mention below, but I think it's a bit nicer to ensure that the `Controller.controller_name` is set when it can be.